### PR TITLE
[ATP-275] Implement croutineExceptionHandler

### PR DESCRIPTION
**Pull request by sdvkit**
Added exception handler

### CODE REVIEW

File: app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt [hunk 1]
```diff
@@ -29,8 +29,7 @@ class DetailsViewModel @Inject constructor(
     private val _state = MutableStateFlow(DetailsState())
     val state: StateFlow<DetailsState> = _state
 
-    private val photoExceptionHandler = CoroutineExceptionHandler { _, throwable ->
-        throwable.printStackTrace()
+    private val photoExceptionHandler = CoroutineExceptionHandler { _, _ ->
```
Comment (VladislavShi) [inline app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt:32]: why no stacktrace?

Comment (sdvkit) [reply app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt:32]: to make any visible changes

Comment (VladislavShi) [reply app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt:32]: yea, but better to do it anyway with visible handling (to see what exactly exception we have in logs). So do both

Comment (sdvkit) [reply app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt:32]: ok

File: app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt [hunk 2]
```diff
@@ -29,8 +30,12 @@ class DetailsViewModel @Inject constructor(
     private val _state = MutableStateFlow(DetailsState())
     val state: StateFlow<DetailsState> = _state
 
-    private val photoExceptionHandler = CoroutineExceptionHandler { _, _ ->
-        _state.value = _state.value.copy(photo = null)
+    private val photoExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+        throwable.printStackTrace()
+
+        if (throwable is UnknownHostException) {
```
Comment (VladislavShi) [inline app/src/main/java/com/sdv/kit/pexelsapp/presentation/details/DetailsViewModel.kt:36]: We should show any exception and print to stacktrace any exception, so we don't need this if

**Pull request merged**